[IAP][UniT] Unit Testing for Storekit Implementation Features

shamin26 opened the pull request:
- Writedown the test cases for didFailWithError Delegate of Storekit
- All cases executed successfully.

Ref : https://redmine.bjitgroup.com/redmine/issues/477111

shamin26 committed:
[IAP][UniT] Unit Testing for Storekit Implementation Features

- Writedown the test cases for didFailWithError Delegate of Storekit
- All cases executed successfully.

Ref : https://redmine.bjitgroup.com/redmine/issues/477111

sadat-399 commented on Tests/EasyPurchaseTests/FetchProductTests.swift:
```diff
@@ -205,4 +213,20 @@ class FetchProductTests: XCTestCase {
+        // Then
+        XCTAssertTrue(fetchProduct.isCompleted)
+        XCTAssertEqual(fetchProduct.cachedProducts?.error as NSError?, mockError)
+
```
You can also add this case.
```
XCTAssertNil(fetchProduct.retrievedProducts)
XCTAssertNil(fetchProduct.invalidProductIDs)
```

shamin26 replied:
Added. Thanks.

shamin26 committed:
[IAP][UniT] Unit Testing for Storekit Implementation Features

- Writedown the test cases for didFailWithError Delegate of Storekit
- All cases executed successfully.

Update:
- Added two new assertions.

Ref : https://redmine.bjitgroup.com/redmine/issues/477111
